.changeset/green-moose-fetch.md

```diff
@@ -0,0 +1,5 @@
+---
+'astro': patch
+---
+
+fix: make isPromise work on proxy objects
```

Comment:
```suggestion
Fixes a case where a `Promise` on a `Proxy` would not be handled correctly
```
I'm not sure about the wording "on a" mainly because idk much about proxies. Any suggestions?